rfcs/text/0030-linux-event-model.md

```diff
@@ -0,0 +1,1239 @@
+
+<!--
+Stage 2: Add or update all remaining field definitions. The list should now be exhaustive. The goal here is to validate the technical details of all remaining fields and to provide a basis for releasing these field definitions as beta in the schema. Use GitHub code blocks with yml syntax formatting, and add them to the corresponding RFC folder.
+-->
+
+## Usage
+
+<!--
+Stage 1: Describe at a high-level how these field changes will be used in practice. Real world examples are encouraged. The goal here is to understand how people would leverage these fields to gain insights or solve problems. ~1-3 paragraphs.
+-->
+
+### Session View
+
+The primary use case for this data will be to drive the "Session view". A time ordered rendering of a Linux session process tree. The addition of nested process ancestors entry_leader, session_leader, and group_leader will allow Session View to load only the events for a particular session.
+
+KQL examples:
+```
+process.entry_leader.entity_id: &lt;entity_id of entry session leader>
+
+or
+
+process.session_leader.entity_id: &lt;entity_id of session leader>
+```
+
+#### Session/Process interactivity (is the session connected to a controlling tty?)
+
+process.tty will be used to determine if the session is interactive. If the field is unset there is no controlling tty and the session is non-interactive (possibly a service). The *process.interactive* boolean will indicate if the process itself is connected to the controlling tty.
+
+#### Pipes and redirects
+
+*process.group_leader* as well as *process.file_descriptions* will allow the Session View to properly represent pipes and redirects in a familiar shell like way. e.g
+`cat /etc/hosts | grep google > somefile.txt`
+
+### Unique process.entity_id generation
+
+*host.boot.id* and *host.pid_ns_ino* will be used in generating unique *entity_id*s for the process using fn(process.pid, process.start, host.pid_ns_ino, host.boot.id)
+
+### Rule engines
+
+The new nested ancestor processes will provide rule engines a widened context to allow for more targeted alerting.
+
+KQL example:
+
+Boot a user from an ssh session if the user is connecting with root and trying to run mysql:
+`process.entry_leader.user.name: root AND process.executable: /bin/mysql`
+
+## Source data
+
+<!--
+Stage 1: Provide a high-level description of example sources of data. This does not yet need to be a concrete example of a source document, but instead can simply describe a potential source (e.g. nginx access log). This will ultimately be fleshed out to include literal source examples in a future stage. The goal here is to identify practical sources for these fields in the real world. ~1-3 sentences or unordered list.
+-->
+
+The source of data will be via kprobes and ebpf sensors. All process execution lifecycle events will be sent up (fork, exec, end) as well as setsid.
+
+Here is a mock example of these events:
+
+### Fork event
+
+```json
+{
```

Comment:
Shouldn't there be "group" fields in these examples (real, effective, saved, supplemental)